front/app/containers/Authentication/Modal.tsx

```diff
@@ -184,6 +184,27 @@ const AuthModal = ({ setModalOpen }: ModalProps) => {
     ? appConfiguration?.data.attributes.settings.core[helperTextKey]
     : undefined;
 
+  // If the user is verifying their identity, save the successAction and
+  // the current context in local storage for when they return.
+  if (
+    currentStep === 'missing-data:verification' ||
+    currentStep === 'verification-only' ||
+    currentStep === 'sign-up:verification'
+  ) {
+    if (authenticationData?.successAction) {
+      localStorage.setItem(
+        'auth_success_action',
+        JSON.stringify(authenticationData.successAction)
+      );
+    }
+    if (authenticationData?.context) {
+      localStorage.setItem(
+        'auth_context',
+        JSON.stringify(authenticationData.context)
+      );
+    }
+  }
```

Comment:
This seemed like the correct place to set these items in local storage for 3rd party verification, but let me know if this is incorrect :slightly_smiling_face:

Reply:
(also - we need to save the context in local storage as well here, since verification doesn't return the same URL params as SSO to reconstruct it later on)

Reply:
I think the logic is fine, but if you put it here in the component it's going to run on every re-render. If you want to put it in this component, you should put it in a `useEffect` that only fires when the `currentStep` dependency changes. But IMO it would make a bit more sense in the `useSteps` hook.

However, ideally this logic should be triggered by the particular 'external' verification methods the moment they force us to leave the platform and come back. In your current implementation, it always gets set when you enter the verification step, even if the verification method active does not actually require you to leave the platform. Ideally we would only set this the moment we leave the platform to an external page- similar to how for `SSO` we only do this in the `handleOnSSOClick`.

Ideally we would do this through the `stepConfig` system to keep that as the centralized place to define verification flows, but unfortunately the 'verification' module has its own 'step' system, because I didn't have enough time to also convert that to the new system when I worked on this last year. The nicest solution that I can think of is to pass the `authenticationData` down to the `Verification` component, and then to the `VerificationSteps` component. In there, you can inside of the `onMethodSelected` check if the `selectedMethod` is part of the set of methods that require you leaving the platform, and if so, write to the local storage. But that would require all verification methods to first call that function before redirecting (I quickly checked `VerificationFranceConnectButton` for FranceConnect and there it should be fine, but probably good to check the others too).

Anything that I'm missing?

Reply:
Let me know if you need help with this!

Reply:
`"auth0" | "criipto" | "id_card_lookup" | "bogus" | "bosa_fas" | "clave_unica" | "cow" | "franceconnect" | "gent_rrn" | "nemlog_in" | "oostende_rrn"`

@luucvanderzee Just making these changes now - do you happen to know which of these require leaving the platform and which do not? :thinking:

Reply:
Verification methods that leave the platform:

- auth0
- bosa_fas
- clave_unica
- criipto
- FranceConnect
- nemlog_in
- vienna_saml

You can check this by searching for the `app.components.VerificationModal.buttons` outlet id, seeing the buttons rendered in that outlet, and checking their `handleOnClick` or equivalent method. For example, the `app/modules/commercial/id_franceconnect/components/VerificationFranceConnectButton.tsx` sets `window.location.href` which means you're leaving the platform. On the other hand, `app/modules/commercial/id_oostende_rrn/components/OostendeRrnButton.tsx` just sets the next verification step, which is a form that's built into our platform that doesn't require you leaving our platform.